### PR TITLE
comparing ast typ func

### PR DESCRIPTION
**Pull request by Contentmaudlin**
This function needs to be there for master to compile. Please approve & merge ASAP.

### CODE REVIEW

File: src/frontend/ast.mli [hunk 1]
```diff
@@ -98,3 +98,8 @@ type ns_decl = {
 }
 
 and program = ns_decl list * let_decl list * fn_decl list
+
+
+let compare_ast_typ l r = match(l,r) with
```
Comment (j-hui) [inline src/frontend/ast.mli:103]: Is this for assignment

File: src/backend/semant.ml [hunk 2]
```diff
@@ -351,7 +350,6 @@ let check (ns, globals, functions) =
   check_functions (functions @ (snd flat_ns)) start_env;
 	([], globals @ fst flat_ns, functions @ snd flat_ns)
 
-
```
Comment (j-hui) [inline src/backend/semant.ml:354]: this is great

**Review by j-hui — APPROVED** [review]

File: src/frontend/ast.mli (hunk 1 above)
Comment (Contentmaudlin) [reply src/frontend/ast.mli:103]: Yep

File: src/backend/semant.ml (hunk 2 above)
Comment (Contentmaudlin) [reply src/backend/semant.ml:354]: A much needed change

File: src/frontend/ast.mli [hunk 3]
```diff
@@ -98,3 +98,8 @@ type ns_decl = {
...
+   | (Array(t1, None), Array(t2, Some i)) -> t1=t2
```
Comment (j-hui) [inline src/frontend/ast.mli:104]: this can be written as
```
Array(t1, None), Array(t2, Some _) -> t1=t2

Comment (Contentmaudlin) [reply src/frontend/ast.mli:104]: true

**Pull request merged**